science/gnuradio/Portfile

```diff
@@ -1,458 +1,317 @@
-    # the best.
-
-    configure.cxxflags-append -std=gnu++11
-
-    # when using GCC on older OSX, certain math functions will not be
-    # part of std:: without this macro defined. This macro required
-    # c++11 to be enabled, which is addressed in the next item.
-
-    configure.cxxflags-append -D_GLIBCXX_USE_C99_MATH_TR1
-
-    # enable C/CXX standard extensions (e.g., "gnu++11" and "gnu11"
-    # instead of than "c++11" and "c11"). GNU Radio itself does not
-    # require these extensions, but when compiling with GCC on older
-    # OSX they allow for various various math functions to be in std::
-    # ... so, enable them.
-
-    configure.args-append \
-        -DCMAKE_CXX_EXTENSIONS=ON \
-        -DCMAKE_C_EXTENSIONS=ON
-
-} else {
-
-    # disable C/CXX standard extensions (e.g., "gnu++11" and "gnu11"
-    # rather than "c++11" and "c11"). GNU Radio itself does not
-    # require these extensions, and when compiling with Clang they
-    # don't really do anything anyway; so, just disable them.
-
-    configure.args-append \
-        -DCMAKE_CXX_EXTENSIONS=OFF \
-        -DCMAKE_C_EXTENSIONS=OFF
-
-}
-
-post-destroot {
-
-    # fix residual cmake module install location when
-    # CMAKE_MODULES_DIR does not do the whole trick
-
-    # GNU Radio installs cmake files as "Config*.cmake", which has the
-    # odd behavior of requiring the files to be in a top-level shared
-    # cmake directory; for MacPorts' cmake, these are, apparently:
-    # ${prefix}/lib:${prefix}/share/cmake .  I don't know why ... but
-    # I do not want cmake files installed into the former, so move
-    # them to the latter.
-
-    if {[file exists ${destroot}${prefix}/lib/cmake]} {
-        xinstall -m 755 -d ${destroot}${prefix}/share
-        move ${destroot}${prefix}/lib/cmake \
-            ${destroot}${prefix}/share
-    }
-}
-
-# per user concensus: enable all variants except +jack, +portaudio,
-# +ctrlport, +ctrlport_thrift, +performance_counters,+debug, and
-# +universal.
-
-default_variants +docs +grc +qtgui +uhd +wavelet \
-    +swig +sdl +zeromq
-
-if {${subport} eq "gnuradio" ||
-    ${subport} eq "gnuradio-devel"} {
-    default_variants +logging +log4cpp
-}
+    -DPYTHON_EXECUTABLE=${python_framework_dir}/bin/python${active_python_version} \
+    -DPYTHON_INCLUDE_DIR=${python_framework_dir}/Headers \
+    -DPYTHON_LIBRARY=${python_framework_dir}/Python \
+    -DGR_PYTHON_DIR=${python_framework_dir}/lib/python${active_python_version}/site-packages
+
+# TODO check "macports-libstdc++" or "libstdc++" should not needed anymore, right?
+# disable C/CXX standard extensions (e.g., "gnu++11" and "gnu11"
+# rather than "c++11" and "c11"). GNU Radio itself does not
+# require these extensions, and when compiling with Clang they
+# don't really do anything anyway; so, just disable them.
+configure.args-append \
+    -DCMAKE_CXX_EXTENSIONS=OFF \
+    -DCMAKE_C_EXTENSIONS=OFF
 
 # variants follow
```

Comment:
Let's add a comment above this, something like "# disable all components by default; enable them only if the variant is enabled"